src/build.d

```diff
@@ -1019,13 +1013,64 @@ version(Windows)
         if (!vsInstallPath.exists)
             throw new Exception("Could not locate the Visual Studio installation directory");
 
-        const vcVersionFile = vsInstallPath.buildPath("VC", "Auxiliary", "Build", "Microsoft.VCToolsVersion.default.txt");
+        const vcInstallPath = vsInstallPath.buildPath("VC");
+        if (!vcInstallPath.exists)
+            throw new Exception("Could not locate the Visual C++ installation directory");
+
+        return vcInstallPath;
+    }
+
+    /**
+    Sets certain MSVC environment variables
+
+    Params:
+        vcInstallPath = the absolute path to the MSVC installation directory
+        model         = a string describing the host's model, "64" or "32"
+    */
+    void setVCVars(string vcInstallPath, string model)
+    {
+        const vcVarsFilename = "vcvars" ~ model ~ ".bat";
+        const vcVars = vcInstallPath.buildPath("Auxiliary", "Build", vcVarsFilename);
+        if (!vcVars.exists)
+            throw new Exception("Could not locate the Visual C++ " ~ vcVarsFilename ~ " file");
+
+        // run vcvars{arch}.bat and return the resulting environment variables with the `set` command
+        const vsVars = ["cmd", "/C", vcVars, "&&", "set"].execute.output;
+
+        // copy the environment variables we need to the current environment
+        const varsToCopy = ["INCLUDE", "LIBPATH"];
```

Comment:
For compilation INCLUDE should be enough, but if you want to link aswell, you need LIB, not LIBPATH. You will probably also need to get some updates to PATH as some DLLs might not be found, e.g. with /DEBUG.

Why not just copy the full environment?

Reply:
> Why not just copy the full environment?

I don't know if there will be conflicts, and I just want to avoid problems and noise.